src/core/Mixer.cpp

```diff
@@ -825,5 +825,41 @@ void Mixer::validateChannelName( int index, int oldIndex )
 	}
 }
 
+bool Mixer::isChannelInUse(int index)
+{
+	// check if the index mixer channel receives audio from any other channel
+	if (!m_mixerChannels[index]->m_receives.isEmpty())
+	{
+		return true;
+	}
+
+	// check if the destination mixer channel on any instrument or sample track is the index mixer channel
+	TrackContainer::TrackList tracks;
+	tracks += Engine::getSong()->tracks();
+	tracks += Engine::patternStore()->tracks();
+
+	for (Track* t : tracks)
```

Comment:
Could use ``auto`` here.
```suggestion
	for (const auto t : tracks)
```